app/services/rate_limit_checker.rb

```diff
@@ -66,6 +66,8 @@ def limit_by_email_recipient_address(address)
   end
 
   def limit_cache_key(action)
+    raise "Invalid Cache Key: user ID can't be blank" unless @user.id
```

Comment:
Quick question, how does this happen in the actual app? I can see `@user` being `nil`, but under that circumstances would we hit the case of the user being present but not having an ID (which is the primary key)?

Reply:
Curious as well 👍

Reply:
I discovered this problem when I implemented a rate limit on email confirmations #7780 and added it to a validate which happens when a user is created which means we pass a new user object to the rate limiter.

Reply:
I see, thanks for clarifying.

The way I most commonly see this check performed is with `@user.persisted?` since that makes it more obvious that we're dealing with a new record.